File: spec/factories/repository.rb
FactoryBot.define do
  factory :repository do
    sequence(:id) { |n| n + 1 }
    name { 'gitstar-ranking' }
    full_name { 'k0kubun/gitstar-ranking' }
    association :owner, factory: :user
    description { 'GitHub Ranks' }
    fork { false }
    homepage { 'http://githubranking.com' }
    stargazers_count { 11 }
    language { 'Ruby' }
    fetched_at { nil }
    rank { 131197 }
  end
end

File: worker/src/main/kotlin/com/github/k0kubun/gitstar_ranking/core/Repository.kt
package com.github.k0kubun.gitstar_ranking.core

data class Repository(
    val id: Long,
    val ownerId: Int? = null, // TODO: should be Long
    val name: String? = null, // TODO: can we remove this to save storage?
    val fullName: String? = null,
    val description: String? = null,
    val fork: Boolean? = null,
    val homepage: String? = null,
    val stargazersCount: Int,
    val language: String? = null,
)
